src/providers/local/join-pool.provider.ts

```diff
@@ -173,17 +196,38 @@ const provider = (props: Props) => {
     });
   }
 
+  /**
+   * Resets previous joinQuery results
+   */
+  function resetState() {
+    bptOut.value = '0';
+    priceImpact.value = 0;
+    queryError.value = '';
+  }
+
   /**
    * Simulate join transaction to get expected output and calculate price impact.
    */
   async function queryJoin() {
+    // If form is empty, clear the price impact and
+    // return early
+    if (!hasAmountsIn.value) {
+      priceImpact.value = 0;
+      return;
+    }
+    // Early return if relayer not yet approved
+    if (shouldSignRelayer.value) return;
```

Comment:
Actually, I don't think we should have this block here. I'm not sure where we should have it but this means for all pool types regardless of whether you need the relayer signature or not, it will block queries. I guess the block should happen based on the join handler type.

Maybe we should have a static property on the JoinHandler interface called `isRelayerSigRequired`. Then for all pool handlers, you have to specify whether it is or not. Then this conditional would be `joinPoolService.isRelayerSigRequired && !hasSignedRelayer.value`